civis/base.py

```diff
@@ -220,10 +224,27 @@ def _civis_state(self):
     @property
     def _state(self):
         """State of the CivisAsyncResultBase in `future` language."""
+        if self.__state is not None:
+            return self.__state
         with self._condition:
             return STATE_TRANS[self._civis_state]
 
     @_state.setter
     def _state(self, value):
         # Ignore attempts to set the _state from the `Future` superclass
-        pass
+        # TODO explain
+        self.__state = value
```

Comment:
this part is weird. I'm going to try to find a way to avoid changing the `_state` setter here